File: 프로그래머스/lv2/12941. 최솟값 만들기/최솟값 만들기.java
import java.util.*;
class Solution
{
    public int solution(int []A, int []B)
    {
        int answer = 0;

        Arrays.sort(A);
        Arrays.sort(B);
        
        int len = A.length;
        int order =0;
        int preOrder =0;
        for(int i=0; i<len; i++){
            order += A[i]*B[len-1-i];
            preOrder += B[i]*A[len-i-1];
        }
        return Math.min(order,preOrder);
    }
}

File: 프로그래머스/lv2/12941. 최솟값 만들기/최솟값 만들기.cpp
#include <iostream>
#include<vector>
#include<algorithm>
using namespace std;

bool com (int a, int b){
    if(a>b) return true;
    return false;
}

int solution(vector<int> A, vector<int> B)
{
    sort(A.begin(),A.end());
    sort(B.begin(),B.end(),com);
    int answer =0;
    for(int i=0; i<A.size(); i++){
        answer += A[i]*B[i];
    }

    return answer;
}